src/dialogs.ts

```diff
@@ -64,3 +65,51 @@ export async function askShouldDownload(): Promise<boolean> {
 
   return false;
 }
+
+export async function askSelectRootDir(): Promise<boolean> {
+  const selectRootDir = extensionConfiguration(SettingsKey.selectRootDir);
+
+  if (!selectRootDir) return false;
+
+  enum Options {
+    yes = "Yes",
+    no = "No",
+    never = "Never",
+  }
+
+  const response = await vscode.window.showInformationMessage(
+    "Multiple Meson projects detected, select one?",
+    ...Object.values(Options),
+  );
+
+  switch (response) {
+    case Options.yes:
+      return true;
+    case Options.no:
+      return false;
+    case Options.never:
+      extensionConfigurationSet(SettingsKey.selectRootDir, false, vscode.ConfigurationTarget.Workspace);
+      return false;
+  }
+
+  return false;
+}
+
+export async function selectRootDir(rootDirs: string[]): Promise<string | undefined> {
+  // TODO: What label to use when there is more than one workspace?
```

Comment:
What shoule we do with this TODO?

Reply:
I think this is the use case:

![obraz](https://github.com/mesonbuild/vscode-meson/assets/600573/a0847185-d2e7-4d72-a83e-e3b5b6a5b775)

The `meson-test-build-multi` is a root of the workspace, and its subdirectory `build` is another root, as you can see in the explorer pane on the right. The subdirectory `and/this/is` has also been added as another root. Finally, the external `meson-test-repo-single` directory has been added as yet another root.

So far, the only problem I see here is that the root directory `meson.build` is left unlabeled. Maybe we should use `/` instead?

---

Another issue is that the list of available configurations is only created when the extension is initialized. If you add VS Code provides an `onDidChangeWorkspaceFolders` event, we should probably listen to it and react accordingly.

Reply:
Ok, turns out there is actually a method specifically for our use-case: `vscode.workspace.asRelativePath()`. I also added back the `/meson.build` suffix as in the original code because that fix 2 problems at once: no more empty entries, and asRelativePath() was returning absolute path when the path is a workspace root.

Reply:
This is the list I get for the above scenario, after the changes. Note that the root `meson.build` is no longer the default on the list.

![obraz](https://github.com/mesonbuild/vscode-meson/assets/600573/78f1d6a3-0c05-4dda-b7f4-664abce972a5)

Reply:
> Another issue is that the list of available configurations is only created when the extension is initialized

Indeed, fixed by calling `mesonRootDirs()` again when running the command to get the new list. I don't think we need to be notified for this.

Reply:
> Indeed, the sorting is broken, but that's handled in https://github.com/mesonbuild/vscode-meson/pull/199, so I think we can leave that out of scope here.

I'm not sure this fix will work properly. It counts the number of directories in a path and sorts by that, but with these new changes, the root `meson.build` no longer has zero directories in its path*. I think it may get mixed up with the rest of the found files.

For example, if I add a new `contrib` workspace root that has `meson.build` at its root, I get this:

![obraz](https://github.com/mesonbuild/vscode-meson/assets/600573/234ca84d-44e2-40ca-8fc5-7070254e238d)

And now both `contrib/meson.build`** and `meson-test-repo-multi/meson.build` are at the same directory depth, and if sorted by name, the contrib one will show up first in the list.

*) The suggested solution of using `/` instead would also break this.
**) Actually `meson-test-repo-multi/contrib/meson.build`.

Reply:
TBH, I haven't looked much what that other PR does, but we can discuss there implementation details.

Reply:
Note that sorting should happen before appending `/meson.build` suffix, that part is really just a display label thing. I think that way it will work.

Reply:
The problem is that `vscode.workspace.asRelativePath()` doesn't seem to return the root level `meson.build` as a null path, instead returning the workspace root name. If roots are to be treated as equivalent, there's not much we can do here, since there is no "primary" root to select.

In the previous approach, the first root was special because all paths were computed relative to it, giving the first root `meson.build` preference.